### DISCUSSION TITLE
Can we frequently visit monitor http server?

### DISCUSSION THREAD

**Discussion by wubumihuo** [nats-io/nats-server#4787] [Q&A] [answered]
we have met several production problems, so we want to get each connection detailed  monitoring info such as connSubscriptions，pendingbytes. can we visit and update monitor http server for 1 time/sec?

**Comment by derekcollison** [comment, marked as answer]
Yes you can access /connz, or using the NATS api to get the same data from a system user.

If you have production issues, you might consider a commercial relationship with Synadia.

**Reply by wubumihuo** [reply]
Thanks, That is to say,  access /connz for tps 1 will not make extraordinarily large performance pressure  for nats-server？
and our production issues was caused by Slowconsumer and large subscriptions ，we have idea how to solve it and we want to early find problem by monitor

**Reply by derekcollison** [reply]
How many connections?

**Reply by derekcollison** [reply]
Also what server version?

**Reply by wubumihuo** [reply]
we use version 2.8.4； our connections is about 500

**Reply by derekcollison** [reply]
Wow.. Try installing the latest server which is 2.10.5. 

But yes that should be ok, but would use the API vs HTTP.

**Reply by wubumihuo** [reply]
Thanks. We mainly use java-client, does API support java?

**Reply by derekcollison** [reply]
Most folks use the NATS cli, but anything that can be done in NATS cli you can add `--trace` to the end and it will show the request and the request subject and the reply of everything it is doing. NATS cli also has schemas which you can use to marshal and unmarshal JSON.
